### PR TITLE
RDM-1944: Initial commit

### PR DESCRIPTION
**Pull request by Daniel-Lam**
Initial version of Case Admin Web, implementing existing Case Definition import functionality.

### JIRA link ###
https://tools.hmcts.net/jira/browse/RDM-1944

### Change description ###
Implementation of import functionality provided by existing Case Definition Import web application.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by Daniel-Lam** [comment]
@timja-kainos - ah yes, I should've done :)

File: src/main/server.ts [hunk 1]
```diff
@@ -1,27 +1,12 @@
 #!/usr/bin/env node
-import * as logging from "@hmcts/nodejs-logging";
-import * as fs from "fs";
-import * as https from "https";
-import * as path from "path";
+import { Logger } from "@hmcts/nodejs-logging";
 import { app } from "./app";
```
Comment (adr1ancho) [inline src/main/server.ts]: why Logger is in upper case

Comment (Daniel-Lam) [reply src/main/server.ts]: It's the named export, see: https://github.com/hmcts/nodejs-logging/blob/master/index.js#L4

Comment (adr1ancho) [reply src/main/server.ts]: I was on the train when typing. Thx for link

File: src/main/app.ts [hunk 2]
```diff
@@ -51,22 +61,24 @@ if (config.useCSRFProtection === true) {
...
 // returning "not found" page for requests with paths not resolved by the router
-app.use((req, res, next) => {
+app.use((req, res, next) => { // eslint-disable-line no-unused-vars
   res.status(404);
   res.render("not-found");
 });
 
 // error handler
-app.use((err, req, res, next) => {
+app.use((err, req, res, next) => { // eslint-disable-line no-unused-vars
```
Comment (tchow8) [inline src/main/app.ts]: I think if you aren't using next here but am using res, you only need to do the params up to that point i.e. (err, req, res). Then you don't need the eslint-disable I believe.

File: src/main/modules/helmet/index.ts [hunk 3]
```diff
@@ -54,10 +54,10 @@ export class Helmet {
     app.use(helmet.referrerPolicy({policy}));
   }
 
-  private setHttpPublicKeyPinning(app, hpkpConfig) {
-    app.use(helmet.hpkp({
-      maxAge: hpkpConfig.maxAge,
-      sha256s: hpkpConfig.sha256s,
-    }));
-  }
+  // private setHttpPublicKeyPinning(app, hpkpConfig) {
```
Comment (tchow8) [inline src/main/modules/helmet/index.ts]: Any reason these are commented? Maybe just a short description above.

File: src/main/user/user-request-authorizer.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  let user;
+  const bearerToken = request.get(AUTHORIZATION) || (request.cookies ? request.cookies[COOKIE_ACCESS_TOKEN] : null);
+
+  if (!bearerToken) {
+    return Promise.reject(ERROR_TOKEN_MISSING);
+  }
+
+  // Use AccessToken cookie as Authorization header
+  if (!request.get(AUTHORIZATION) && bearerToken) {
+    if (!request.headers) {
```
Comment (tchow8) [inline src/main/user/user-request-authorizer.ts]: Might look better like:
request.headers = request.headers || {};
request.headers[AUTHORIZATION] = `Bearer ${bearerToken}`;
Preference really.

File: src/main/app.ts (hunk 2 above)
Comment (Daniel-Lam) [reply src/main/app.ts]: Not quite correct... I did try this out but it didn't work. Express requires you explicitly pass _four_ arguments for this middleware to be treated as an error handler. See https://expressjs.com/en/guide/error-handling.html

File: package.json [hunk 5]
```diff
@@ -29,34 +33,74 @@
...
+    "jsonwebtoken": "^8.2.1",
+    "karma": "1.4.1",
     "mocha": "^3.5.3",
     "mochawesome": "^2.3.1",
-    "nock": "^9.0.16",
+    "nock": "^9.1.4",
     "nsp": "^2.8.0",
-    "pa11y": "^4.13.0",
+    "nyc": "^11.8.0",
+    "pa11y": "^4.13.2",
```
Comment (adr1ancho) [inline package.json]: should we move nyc above line pally

**Pull request merged**